hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/virtual/ContractKey.java

```diff
@@ -178,12 +178,17 @@ public int serializeReturningByteWritten(final SerializableDataOutputStream out)
     }
 
     @Override
-    public void serialize(final ByteBuffer byteBuffer) throws IOException {
-        byteBuffer.put(getContractIdNonZeroBytesAndUint256KeyNonZeroBytes());
+    public void serialize(final ByteBuffer buffer) throws IOException {
+        serializeReturningBytesWritten(buffer);
+    }
+
+    public int serializeReturningBytesWritten(final ByteBuffer buffer) {
+        buffer.put(getContractIdNonZeroBytesAndUint256KeyNonZeroBytes());
         for (int b = contractIdNonZeroBytes - 1; b >= 0; b--) {
-            byteBuffer.put((byte) (contractId >> (b * 8)));
+            buffer.put((byte) (contractId >> (b * 8)));
         }
-        serializePackedBytesToBuffer(uint256Key, uint256KeyNonZeroBytes, byteBuffer);
+        serializePackedBytesToBuffer(uint256Key, uint256KeyNonZeroBytes, buffer);
+        return 1 + contractIdNonZeroBytes + uint256KeyNonZeroBytes;
```

Comment:
It might be worth mentioning the reason for adding 1 here.

Reply:
Serialized form is:
* 1 byte that indicates total number of non-zero bytes
* followed by contract ID non-zero bytes
* followed by unit256key non-zero bytes

Let me add a few comments to make it more clear.

Reply:
Serialized form is:
* 1 byte that indicates total number of non-zero bytes
* followed by contract ID non-zero bytes
* followed by unit256key non-zero bytes

Let me add a few comments to make it more clear.

Reply:
Thank you !